packages/emitter-framework/src/python/components/source-file.tsx

```diff
@@ -0,0 +1,78 @@
+import { Children } from "@alloy-js/core/jsx-runtime";
+import {
+  createContext,
+  OutputSymbol,
+  reactive,
+  Scope,
+  SourceFile as CoreSourceFile,
+} from "@alloy-js/core";
+import { PythonOutputSymbol, Reference } from "../index.js";
+import { ImportStatements, ImportSymbol } from "./import-statement.jsx";
+import { usePackage } from "./python-package.jsx";
+
+export interface SourceFileContext {
+  addImport(symbol: OutputSymbol): string;
+}
+
+export const SourceFileContext = createContext<SourceFileContext>();
+
+export interface SourceFileProps {
+  path: string;
+  children?: Children;
+}
+
+/**
+ * Represents a Python source file.
+ *
+ * Handles top level package declaration, as well as importing other sources
+ */
+export function SourceFile(props: SourceFileProps) {
+  const packageCtx = usePackage();
+
+  if (!packageCtx) {
+    throw new Error("SourceFile must be declared inside a package");
+  }
+
+  // Collection of import symbols
+  const importRecords: ImportSymbol[] = reactive([]);
+  // Map a symbol to import name, keep track of already imported symbols
+  const importedSymbols = new Map<OutputSymbol, string>();
+
+  // Add import to file if not already, returns name of imported symbol
+  function addImport(symbol: PythonOutputSymbol): string {
+    if (importedSymbols.has(symbol)) {
+      return importedSymbols.get(symbol)!;
+    }
+
+    // Only need to import if not in same package
+    if (symbol.package !== packageCtx?.qualifiedName) {
```

Comment:
I think this will create multiple import statements for the same package? Is that expected?

Reply:
>  Is that expected?

Probably not--I basically copied this from the Java implementation.